src/Sylius/Bundle/CoreBundle/Form/Type/ChannelCollectionType.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\CoreBundle\Form\Type;
+
+use Sylius\Component\Channel\Repository\ChannelRepositoryInterface;
+use Sylius\Component\Core\Model\ChannelInterface;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * @author Kamil Kokot <kamil.kokot@lakion.com>
+ */
+final class ChannelCollectionType extends AbstractType
+{
+    /**
+     * @var ChannelRepositoryInterface
+     */
+    private $channelRepository;
+
+    /**
+     * @param ChannelRepositoryInterface $channelRepository
+     */
+    public function __construct(ChannelRepositoryInterface $channelRepository)
+    {
+        $this->channelRepository = $channelRepository;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder->addEventListener(FormEvents::PRE_SET_DATA, function (FormEvent $event) use ($options) {
+            $form = $event->getForm();
+
+            foreach ($form as $name => $child) {
+                $form->remove($name);
+            }
+
+            /** @var ChannelInterface[] $channels */
+            $channels = $this->channelRepository->findAll();
+            foreach ($channels as $channel) {
+                $form->add($channel->getCode(), $options['entry_type'], array_replace([
+                    'property_path' => '[' . $channel->getCode() . ']',
+                ], $options['entry_options'], [
+                    'block_name' => 'entry',
+                    'channel' => $channel,
+                ]));
+            }
+        });
+    }
+
+    /**
+     * @{@inheritdoc}
+     */
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        $resolver->setRequired([
+            'entry_type',
+        ]);
+
+        $resolver->setDefaults([
+            'error_bubbling' => false,
+            'entry_options' => [],
+        ]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getBlockPrefix()
+    {
+        return 'collection';
```

Comment:
To reuse collection templates (it does not extend `CollectionType`, but it behaves like it (with options `allow_add`, `allow_delete`, `delete_empty` set to false).